packages/ui/pod-pcd-ui/src/CardBody.tsx

```diff
@@ -1,26 +1,97 @@
+import { Button, Separator } from "@pcd/passport-ui";
 import { PCDUI } from "@pcd/pcd-types";
-import { PODPCD } from "@pcd/pod-pcd";
+import { PODPCD, PODPCDPackage } from "@pcd/pod-pcd";
+import { useState } from "react";
 import { CollectablePODPCDCardBody } from "./renderers/CollectablePODPCDCardBody";
 import { DefaultPODPCDCardBody } from "./renderers/DefaultPODPCDCardBody";
+import { Container } from "./shared";
 
 export const PODPCDUI: PCDUI<PODPCD> = {
   renderCardBody: PODPCDCardBody
 };
 
+const DEF_POD_DISPLAY_FORMATS = ["pod", "card", "collectable"] as const;
+type PODDisplayFormat = (typeof DEF_POD_DISPLAY_FORMATS)[number];
+const ALL_POD_DISPLAY_FORMATS: string[] = [...DEF_POD_DISPLAY_FORMATS];
+const DISPLAY_FORMAT_NAMES = {
+  pod: "POD",
+  collectable: "Card",
+  card: "Card"
+};
+
 /**
  * This component renders the body of a 'Card' that Zupass uses to display PCDs to the user.
  */
 function PODPCDCardBody({ pcd }: { pcd: PODPCD }): JSX.Element {
-  let content = <></>;
+  const [sigStatus, setSigStatus] = useState<number>(0);
 
-  switch (pcd.claim.entries["zupass_display"]?.value) {
+  const availableDisplayFormat = getPreferredDisplayFormat(pcd);
+  const [displayFormat, setDisplayFormat] = useState<PODDisplayFormat>(
+    availableDisplayFormat || "pod"
+  );
+  const otherDisplayFormat =
+    displayFormat === "pod" ? availableDisplayFormat : "pod";
+
+  let content = <></>;
+  switch (displayFormat) {
     case "collectable":
       content = <CollectablePODPCDCardBody pcd={pcd} />;
       break;
+    case "pod":
+    // Fallthrough
     default:
       content = <DefaultPODPCDCardBody pcd={pcd} />;
       break;
   }
 
-  return <>{content}</>;
+  return (
+    <Container>
+      {content}
+
+      <Separator />
+      {otherDisplayFormat === undefined ? null : (
+        <Button
+          style="secondary"
+          size="small"
+          onClick={async (): Promise<void> =>
+            setDisplayFormat(otherDisplayFormat || "pod")
+          }
+          styles={{ float: "left" }}
+        >
+          {otherDisplayFormat === undefined
+            ? "No display info"
+            : `View as ${DISPLAY_FORMAT_NAMES[otherDisplayFormat]}`}
+        </Button>
+      )}
+
+      <Button
+        style={
+          sigStatus === 0 ? "primary" : sigStatus > 0 ? "outline" : "danger"
+        }
+        size="small"
+        onClick={async (): Promise<void> =>
+          setSigStatus((await PODPCDPackage.verify(pcd)) ? 1 : -1)
```

Comment:
there's a small chance `verify` could throw, in which we should probably display a verification error?